### PR TITLE
Devx 3585 cleanup

### PR DESCRIPTION
**Pull request by igorwojda**
https://nexmoinc.atlassian.net/browse/DEVX-3585

Clean up sample apps
- Add new callbacks
- Clean up code
- Upgrade dependencies
- Upgrade AGP
- Upgrade Gradle

### CODE REVIEW

File: AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/MainActivity.java [hunk 1]
```diff
@@ -651,6 +673,11 @@ public void onVideoQualityWarningLifted(OTWrapper otWrapper, String remoteId) th
                     Log.i(LOG_TAG, "The quality has improved");
                 }
 
+                @Override
+                public void onAudioLevelUpdated(float v) throws ListenerException {
+
+                }
+
```
Comment (Raulkg) [inline AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/MainActivity.java:680]: Any comment or log that explains this method and necessity of  float value  could be helpful.

File: AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java [hunk 2]
```diff
@@ -213,16 +212,16 @@ public void updateCall() {
             mControlCallbacks.onCall();
     }
 
-    private void updateScreensharing() {
-        mVideoBtn.setOnClickListener(!mActivity.isScreensharing()
+    private void updateScreenSharing() {
```
Comment (Raulkg) [inline AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java:215]: I have noticed different access modifiers for the same access in the anonymous class

- updateLocalAudio() - public void 
- updateLocalVideo() - public void 
- updateCall() - public void 
- updateScreensharing() - private void 
- updateAnnotations()  - private void 
- updateTextChat()  - private void 

If there is no reason for them to be like that, i believe they should all have same access.

Comment (igorwojda) [reply AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java:215]: Some of them are used directly in the `MainActivity` and thus they have to be `public`. We could align access modifiers and make all of them public, but exposing more then is needed is a bad practice.

File: AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/MainActivity.java [hunk 3]
```diff
@@ -248,35 +248,33 @@ public boolean onCreateOptionsMenu(Menu menu) {
...
-                        }
-                    });
-                    builder.setNegativeButton("RE-TRY", new DialogInterface.OnClickListener() {
-                        @Override
-                        public void onClick(DialogInterface dialog, int which) {
-                            dialog.dismiss();
-                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                                requestPermissions(permissions, permsRequestCode);
-                            }
+        if (permsRequestCode == 200) {
```
Comment (Raulkg) [inline AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/MainActivity.java:251]: Can this be a constant.

i mean it is confusing to have a constant with name "permsRequestCode" and validate different variable with same name against constant value. 

Note:  rename "permsRequestCode" to PERMISSION_CODE 

Example:

```
public void onRequestPermissionsResult(final int requestCode, final String[] permissions,
                                           int[] grantResults) {	                                           int[] grantResults) {
        if (requestCode == PERMISSION_CODE ) {
....
```

File: AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java (hunk 2 above)
Comment (igorwojda) [reply AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java:215]: I have changed few of them (where it was possible)
https://github.com/opentok/accelerator-sample-apps-android/pull/23/commits/7920502af2a916881d10debb9bcacc7cfa2c06b1

I think we should keep rest as it is

File: AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/MainActivity.java (hunk 3 above)
Comment (igorwojda) [reply AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/MainActivity.java]: done

File: AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java (hunk 2 above)
Comment (Raulkg) [reply AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java:215]: I just checked main activity and found no usages for public methods below: 

updateLocalAudio() - public void
updateLocalVideo() - public void
updateCall() - public void

**Review by Raulkg — APPROVED** [review]
LGTM

File: AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java (hunk 2 above)
Comment (igorwojda) [reply AcceleratorSampleApp/app/src/main/java/com/tokbox/android/accelerator/sample/ui/ActionBarFragment.java:215]: made them private

**Review by marinaserranomontes — APPROVED** [review]
LGTM

**Pull request merged**